packages/flutter_tools/test/integration.shard/test_flutter_tree.dart

```diff
@@ -0,0 +1,391 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/file.dart';
+
+import '../src/common.dart';
+import '../src/flutter_tree.dart';
+import '../src/process.dart';
+import 'test_utils.dart';
+
+/// The Flutter source tree that this program is part of.
+///
+/// The tree is the one located by [getFlutterRoot].
+final FlutterTree hostFlutterTree = FlutterTree(
+    fileSystem.directory(getFlutterRoot()).absolute);
+
+/// The value the entrypoint writes into `flutterToolsStampFile`.
+///
+/// The [toolArgs] parameter corresponds to `FLUTTER_TOOL_ARGS`
+/// in the entrypoint scripts.
+String flutterToolsStampValue({required String revision, required String toolArgs}) {
+  return '$revision:$toolArgs';
+}
+
+/// Additional members on [FlutterTree] which are helpful for [TestFlutterTree]
+/// and its users.
+extension FlutterTreeExtension on FlutterTree {
```

Comment:
Can you make these non-extension helper methods? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension